include/sys/dmu.h

```diff
@@ -480,7 +480,7 @@ int dmu_object_set_blocksize(objset_t *os, uint64_t object, uint64_t size,
  * to accommodate the change.
  */
 int dmu_object_set_maxblkid(objset_t *os, uint64_t object, uint64_t maxblkid,
-    dmu_tx_t *tx);
+    boolean_t force, dmu_tx_t *tx);
```

Comment:
since this is always called with `force=TRUE` (and only from receive_object()), I don't think we need the `force` argument.  We should update the comment to indicate that it will reduce the maxblkid as well and should be used with care (and will it reduce the nlevels when reducing maxblkid?)

Reply:
> and will it reduce the nlevels when reducing maxblkid?
no. the caller must take care of that first.